sourcegraph/sourcegraph.proto

```diff
@@ -781,6 +786,20 @@ message ChangesetUpdateOp {
 	UserSpec Author = 8 [(gogoproto.nullable) = false];
 }
 
+message ChangesetUpdateAffectedOp {
+	// Repo holds the RepoSpec which received a commit.
+	RepoSpec repo = 1 [(gogoproto.nullable) = false];
+
+	// Branch is the name of the branch which was pushed to.
+	string branch = 2;
+
+	// Last is the SHA1 of the previous commit on the branch.
+	string last = 3;
+
+	// Commit is the SHA1 of the newly pushed commit on the branch.
+	string commit = 4;
```

Comment:
Are both current and last commits necessary? Can you provide motivation for having both? (I'm not objecting it, I just want to understand.)